lib/Doctrine/DBAL/Platforms/MySqlPlatform.php

```diff
@@ -460,12 +461,23 @@ protected function _getCreateTableSQL($tableName, array $columns, array $options
     }
 
     /**
+     * Tells whether a field type supports declaration of a default value.
+     *
+     * MySQL (as of 5.7.19) does not support default values for Blob and Text
+     * columns while MariaDB 10.2.1 does.
+     */
+    protected function isDefaultValueSupportedForType(Type $field): bool
```

Comment:
Thanks !!! This was a question I asked 2 weeks ago without feedback and here it is ;)

> In the current implementation, MySqlPlatform strips default values for BLOB and TEXT columns. It was quite coupled so I ended up creating a special method MySqlPlatform::isDefaultValueSupportedForType(Type $type). Not very happy about it... I though about different implementations: create a supportsBlobTextDefault... but ultimately it's better to get your feedback on this.

Yes it would be more clean. I'll add a separate comment about this.